yt/visualization/plot_window.py

```diff
@@ -876,25 +873,54 @@ def set_axes_unit(self, unit_name):
     @invalidate_plot
     def toggle_right_handed(self):
         issue_deprecation_warning(
-            "toggle_right_handed has been deprecated, use '.flip_horizontal()' instead.",
+            "the toggle_right_handed method is deprecated, use `.flip_horizontal()` instead.",
             since="4.1.0",
             removal="4.2.0",
         )
         self.flip_horizontal()
+        return self
 
     @invalidate_plot
     def flip_horizontal(self):
         """
         flips the horizontal axis (the image's abscissa)
         """
         self._flip_horizontal = not self._flip_horizontal
+        return self
 
     @invalidate_plot
     def flip_vertical(self):
         """
         flips the vertical axis (the image's ordinate)
         """
         self._flip_vertical = not self._flip_vertical
+        return self
+
+    @property
+    def _has_flipped_vertical(self):
+        return self._check_flip_quiver_conflict(self._flip_vertical)
+
+    @property
+    def _has_flipped_horizontal(self):
+        return self._check_flip_quiver_conflict(self._flip_horizontal)
+
+    @property
+    def _has_quiver_callback(self):
+        q_names = ["Quiver", "CuttingQuiver", "MagField", "Velocity"]
+        callback_list = [c[0] for c in self._callbacks]
+        has_q = any([f"{q}Callback" in callback_list for q in q_names])
+        return has_q
+
+    def _check_flip_quiver_conflict(self, flip_axis):
```

Comment:
not sure about this one (which is good motivation to add annotations, actually)
```suggestion
    def _check_flip_quiver_conflict(self, flip_axis:bool) -> bool:
```